Dockerfile

```diff
@@ -1,12 +1,26 @@
 FROM golang:alpine AS builder
-RUN apk --no-cache upgrade
-RUN apk --no-cache add alpine-sdk
-COPY ./ /go/src/github.com/brandond/syslog_ng_exporter/
-WORKDIR /go/src/github.com/brandond/syslog_ng_exporter/
-RUN make
-
-FROM quay.io/prometheus/busybox:latest
-LABEL maintainer="Brad Davidson <brad@oatmail.org>"
-COPY --from=builder /go/src/github.com/brandond/syslog_ng_exporter/syslog_ng_exporter /bin/syslog_ng_exporter
-ENTRYPOINT ["/bin/syslog_ng_exporter"]
-EXPOSE     9577
+
+LABEL maintainer="Jabes Pauya <jabpau93@gmail.com>"
+
+ENV GO1111MODULE=on 
+
+WORKDIR /app
+
+# Copy go mod and sum files
+COPY go.mod .
+
+COPY go.sum .
+
+RUN go mod download
+
+COPY . .
+
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build 
```

Comment:
Why copy the two files separately, and then copy everything for the build?

Does setting GOARCH here break the other work you've done to enable multiarch builds?

Reply:
as i test before this works on my test